### PR TITLE
CA-242017: Add new VMSS class for parsing email content

### PR DESCRIPTION
**Pull request by geosharath**
Added three new classes to parse email subject and body content
for VMSS feature specfic alerts

classes
VmssSnapshotFailed: Handles snapshot schedule failed alerts
VmssSnapshotMissedEvent: Handles snapshot schedule missed event alerts
VmssSnapshotSucceeded: Handles snapshot schedule succeeded alerts

Signed-Off-By: Sharath Babu <sharath.babu@citrix.com>

### CODE REVIEW

File: scripts/mail-alarm [hunk 1]
```diff
@@ -367,6 +367,52 @@ class HAHostFailedETG(EmailTextGenerator):
...
+class VmssSnapshotFailed(EmailTextGenerator):
+    def __init__(self,text):
+        self.text = text
+
+    def generate_subject(self):
+        self.pool_name = get_pool_name()
+        return 'Snapshot schedule failed in pool %s' %self.pool_name
+	
+    def generate_body(self):
+        body = 'The snapshot schedule failed because a snapshot could not be created on the below VM(s): \n'
```
Comment (jonludlam) [inline scripts/mail-alarm:379]: following, not below

File: scripts/mail-alarm [hunk 2]
```diff
@@ -367,6 +367,52 @@ class HAHostFailedETG(EmailTextGenerator):
...
+        body += '\nPlease check %s for more details.' % branding.BRAND_CONSOLE
+        return body
+
+class VmssSnapshotMissedEvent(EmailTextGenerator):
+    def __init__(self,text):
+        self.text = text
+
+    def generate_subject(self):
+        self.pool_name = get_pool_name()
+        return 'Snapshot schedule missed an event in pool %s' %self.pool_name
```
Comment (jonludlam) [inline scripts/mail-alarm:399]: How about 'Scheduled snapshot(s) not taken in pool %s'?

File: scripts/mail-alarm [hunk 3]
```diff
@@ -367,6 +367,52 @@ class HAHostFailedETG(EmailTextGenerator):
...
+        return 'Unable to create snapshots at this time because the scheduled snapshot operation for' \
+                ' this policy is already in progress.'
+
+class VmssSnapshotSucceeded(EmailTextGenerator):
+    def __init__(self,text):
+        self.text = text
+
+    def generate_subject(self):
+        self.pool_name = get_pool_name()
+        return 'Snapshot schedule succeeded in pool %s' %self.pool_name
```
Comment (jonludlam) [inline scripts/mail-alarm:411]: "Scheduled snapshot succeeded"

File: scripts/mail-alarm [hunk 4]
```diff
@@ -367,6 +367,52 @@ class HAHostFailedETG(EmailTextGenerator):
...
+
+    def generate_body(self):
+        return 'Snapshot schedule succeeded.'
```
Comment (jonludlam) [inline scripts/mail-alarm:414]: as above.

**Comment by robhoes** [comment]
@geosharath Could you please update this PR based on the feedback? Also, we'd probably want this to go into the master branch rather than the feature branch, since we've already merged VMSS into master?

**Comment by sharady** [comment]
This PR seems not to be valid now. The CA ticket was closed as won't fix.
VMSS existing mail gives sufficient information, no changes are required.
We can close this PR.

**Comment by robhoes** [comment]
Thanks @sharady. Closing.

**Pull request closed without merging**